### PR TITLE
Open glossary via GUI

### PR DESCRIPTION
**Pull request by ThePhilgrim**
I have chosen to use [customtkinter](https://github.com/TomSchimansky/CustomTkinter) for the GUI in `LinguistLab`. This will make it easier to create a program that looks nice and modern.

In this PR, the main GUI window is created with a sidebar containing a button. The button lets the user open a glossary file via the GUI and that way instantiate the `Glossary` class. 

Mypy is currently ignoring `linguistlab_gui.py` since `customtkinter` does not have any stubs. I have tried to generate stubs for it but I have not yet been successful. As soon as there are stubs for it, the gui file should be checked with mypy in Github actions.

### CODE REVIEW

File: linguistlab.py [hunk 1]
```diff
@@ -1,20 +1,29 @@
 import csv
-import sys
+import pathlib
 
+from tkinter import filedialog
 from typing import Dict, List, Any, Optional
 
 
-def read_glossary(csv_file: Any) -> Dict[str, List[str]]:
+# TODO: Handle wrongly formatted glossary/csv files
```
Comment (Akuli) [inline linguistlab.py:8]:
```suggestion
# TODO: Handle wrongly formatted glossary/csv files and choosing "Cancel" when asking file
```

File: linguistlab.py [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
+def open_glossary() -> "Glossary":
+    path = pathlib.Path(filedialog.askopenfilename())
+
+    if path.suffix != ".csv":
+        raise TypeError("LinguistLab currently only supports glossaries in .csv format.")
+
+    glossary_name = path.name.split(".")[0].title()
```
Comment (Akuli) [inline linguistlab.py:15]:
```
>>> from pathlib import Path
>>> Path("/foo/bar/asdf.txt").stem
'asdf'
```

So you could do:

```suggestion
    glossary_name = path.stem.title()
```

This also handles the corner case where your path starts with a dot, which is typical for hidden files: `.secretglossary.csv`.

Another thing to note is that `.title()` doesn't handle `'s` very well:

```
>>> Path("/home/akuli/akuli's glossary.csv").stem.title()
"Akuli'S Glossary"
```

I personally would just display `path.name` or `path.stem` to the user as is.

**Review by Akuli — COMMENTED** [review]
Since the project is at its early stages, I think the "pull request + review" workflow isn't best. I would recommend committing stuff to main directly, and fixing the bugginess later instead of relying on reviews to find bugs. It is faster, which matters the most when you're trying to kickstart a new project.

That said, a pull request and review is probably a good idea if you have something that is unusually complicated or tricky, for example.

**Comment by ThePhilgrim** [comment]
Thanks for the suggestions! I'll remove `.title()`!

Thanks also for your input about pushing to main. I'll remove the branch protection and continue pushing to main for a while!

**Pull request merged**